src/ImageRunModal.jsx

```diff
@@ -27,13 +27,112 @@ const units = {
+        this.validateAndChange(ports);
+    }
+
+    validateAndChange(ports) {
+        if (typeof this.props.onChange !== 'function')
+            return;
+        let validated = ports.filter(port => parseInt(port.containerPort) >= 1 && parseInt(port.containerPort) <= 65535 &&
+                                     parseInt(port.hostPort) >= 1 && parseInt(port.hostPort) <= 65535);
+        this.props.onChange(validated.map(port => port.hostPort + ':' + port.containerPort + '/' + port.protocol));
+    }
+
+    render () {
+        const dialogValues = this.state;
+        return (
+            <div>
+                {
+                    dialogValues.publish.map((port, idx) => {
+                        return (
+                            <div className='publish-port-form ct-form-layout' key={'published-port-' + idx}>
+                                <div role='group' className='ct-form-layout-split'>
+                                    <input className='form-control'
+                                           type='number'
+                                           step={1}
+                                           min={1}
+                                           max={65535}
+                                           placeholder={_("container port")}
+                                           value={port.containerPort || ''}
+                                           onChange={e => this.onPortChange(idx, 'containerPort', e.target.value)} />
+                                    <Select.Select className='form-control'
+                                                   initial={port.protocol}
+                                                   onChange={value => this.onPortChange(idx, 'protocol', value)} >
+                                        <Select.SelectEntry data='TCP' key='TCP'>
+                                            {_("TCP")}
+                                        </Select.SelectEntry>
+                                        <Select.SelectEntry data='UDP' key='UDP'>
+                                            {_("UDP")}
+                                        </Select.SelectEntry>
+                                    </Select.Select>
+                                    <input className='form-control'
+                                           type='number'
+                                           step={1}
+                                           min={1}
+                                           max={65535}
+                                           placeholder={_("host port")}
+                                           value={port.hostPort || ''}
+                                           onChange={e => this.onPortChange(idx, 'hostPort', e.target.value)} />
+                                </div>
+                                <div role='group' className='ct-form-layout-split'>
+                                    <Button className='btn btn-default pficon-close'
```

Comment:
patternfly offers the bsStyle=default atribute if you want instead of the className=btn btn-default.
If you don't you it you don't need to use pf Buttons.